build_tools/python/e2e_model_tests/run_module_utils.py

```diff
@@ -0,0 +1,58 @@
+## Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Utils that help launch iree run module tools."""
+
+from typing import Any, List
+
+from e2e_test_framework.definitions import common_definitions
+from e2e_test_framework.definitions.iree_definitions import ModuleExecutionConfig, RuntimeDriver
+from e2e_test_framework.device_specs import device_parameters
+
+
+def build_run_flags_for_model(
+    model: common_definitions.Model,
+    model_input_data: common_definitions.ModelInputData) -> List[str]:
+  """Returns the IREE run module flags for the model and its inputs."""
+
+  run_flags = [f"--entry_function={model.entry_function}"]
+  if model_input_data != common_definitions.ZEROS_MODEL_INPUT_DATA:
+    raise ValueError("Currently only support all-zeros data.")
+  run_flags += [
+      f"--function_input={input_type}=0" for input_type in model.input_types
+  ]
+  return run_flags
+
+
+def build_run_flags_for_execution_config(
+    module_execution_config: ModuleExecutionConfig,
+    gpu_id: str = "0") -> List[str]:
+  """Returns the IREE run module flags of the execution config."""
+
+  run_flags = list(module_execution_config.extra_flags)
+  driver = module_execution_config.driver
+  if driver == RuntimeDriver.CUDA:
+    run_flags.append(f"--device=cuda://{gpu_id}")
+  else:
+    run_flags.append(f"--device={driver.value}")
```

Comment:
Hmmm this is ok for now, but feels a bit hacky. Would be nice to figure out a cleaner way to encode this information in the right place.